template/src/api/authSlice.ts

```diff
@@ -0,0 +1,104 @@
+import { PayloadAction, createSlice } from '@reduxjs/toolkit'
+import { REHYDRATE, persistReducer } from 'redux-persist'
+import { PersistPartial } from 'redux-persist/es/persistReducer'
+import { call, put, takeLatest, takeLeading } from 'typed-redux-saga'
+import { Failure, Loading, NotRequested, RemoteData, Success, isSuccess } from '@api/RemoteData'
+import { logIn as logInRequest } from '@api/auth'
+import { AuthTokens, setAuthConfig } from '@api/common'
+import { Credentials } from '@api/types/auth.types'
+import { safeStorage } from '@redux/persistence'
+import { resetStore } from '@redux/rootActions'
+import { RootState } from '@redux/store'
```

Comment:
I think it should not be importing `RootState`, because it causes requirement cycles: `authSlice.ts` imports `@redux/store`, which then again imports this `authSlice.ts`.

I propose to define local `RootState` type here, which will contain only what's used by this `authSlice.ts` and nothing more.

BTW this project is not checking for requirement cycles. Maybe it makes sense to add an issue and take a look at it later in another PR. What do you think?

Reply:
It's not an issue for typescript types. https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-8.html#type-only-imports-and-export

Reply:
we have similar type circular dependency for react navigation screen param types.

Reply:
I have created an issue for this -> https://github.com/brains-and-beards/react-native-template-redbeard/issues/34